### PR TITLE
Document GITHUB_TOKEN workaround

### LINKED ISSUE #1177: GitHub Pages build not triggered with GITHUB_TOKEN
According to https://github.com/JuliaDocs/Documenter.jl/pull/1165 `GITHUB_TOKEN`is sufficient for Documenter to push to `gh-pages`.

However it seems that this token is not allowed to trigger pages builds [1]:
> The `GITHUB_TOKEN` that is provided as a part of GitHub Actions doesn't have authorization to create any successive events, such as GitHub Pages builds. So while you can push to the `gh-pages` branch using the `GITHUB_TOKEN`, it won't spawn a GitHub Pages build. You'll need to create a personal access token and supply it to your GitHub Action as a secret. 

Hence, docs. are correctly pushed, but GitHub Pages are never updated.  
Is there any workaround ?

I tried to add a manual build trigger [2] in my [workflow](https://github.com/maxmouchet/HMMBase.jl/blob/dbeb087f4632a9a8013e5778ab37fa8b141368d2/.github/workflows/documentation.yml):
```yaml
- name: Build and deploy
  env:
    DOCUMENTER_DEBUG: true
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
      run: |
        julia --project=docs/ docs/make.jl
        curl -u maxmouchet:${GITHUB_TOKEN} -X POST https://api.github.com/repositories/164427411/pages/builds -H "Accept: application/vnd.github.mister-fantastic-preview+json"
```
but the GitHub API [returns](https://github.com/maxmouchet/HMMBase.jl/commit/dbeb087f4632a9a8013e5778ab37fa8b141368d2/checks?check_suite_id=307586853#step:5:149) the following error:
```json
{
  "message": "Resource not accessible by integration",
  "documentation_url": "https://developer.github.com/v3/repos/pages/#request-a-page-build"
}
```

[1] https://github.community/t5/GitHub-Actions/Github-action-not-triggering-gh-pages-upon-push/td-p/26869
[2] https://stackoverflow.com/a/54088985

### PR DESCRIPTION
**Pull request by bramtayl**
Closes #1177

### CODE REVIEW

File: docs/src/man/hosting.md [hunk 1]
```diff
@@ -227,15 +231,9 @@ GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
...
-    elsewhere you should instead use a SSH deploy key.
-
-!!! warning "GitHub Pages and GitHub Token"
-    Currently the GitHub Page build is not triggered when the GitHub provided
-    `GITHUB_TOKEN` is used for authentication. See
-    [issue #1177](https://github.com/JuliaDocs/Documenter.jl/issues/1177)
-    for more information.
+    While GitHub will automatically create a github token for you named "GITHUB_TOKEN",
+    it will not have push access. Therefore, you must override
+    GitHub by making a new personal access token with the same name.
```
Comment (aminya) [inline docs/src/man/hosting.md]: It should specifically say what to do.

**Comment by aminya** [comment]
I also think more adjustment to the documentation is needed based on what I said here:
 https://github.com/JuliaDocs/Documenter.jl/issues/1177#issuecomment-586662631

File: docs/src/man/hosting.md (hunk 1 above)
Comment (bramtayl) [reply docs/src/man/hosting.md]: I can add a note like `(see instructions above)` because I did add instructions to do this?

Comment (aminya) [reply docs/src/man/hosting.md]: Yes, and a link to that part of the instructions

**Comment by bramtayl** [comment]
Now that github actions is up and running maybe we should just ditch all the travis and ssh-key docs?

**Comment by aminya** [comment]
> 
> 
> Now that github actions is up and running maybe we should just ditch all the travis and ssh-key docs?

I agree. Probably it is better to move them to a separate document called `Travis.md`.

**Comment by fredrikekre** [comment]
IMO we should not Document this option and only recommend SSH keys (as ww aöready do).

**Comment by bramtayl** [comment]
Why?

**Comment by fredrikekre** [comment]
Because access tokens are not limited to one repo, and is tied to a user, so it would be like giving your github password to everyone with write access to the repo in question.

**Comment by bramtayl** [comment]
Hmm I think I see what your saying. Still seems like a pain to make SSH keys for each individual repo. I wonder if there's another solution?

**Comment by fredrikekre** [comment]
You can reuse it if you want

**Comment by mortenpi** [comment]
I have to agree with @fredrikekre here: we shouldn't recommend the use of personal access tokens for this.

> You can reuse [the SSH deploy key] if you want

I don't think this is true. If you try to add the same deploy key again, GitHub will complain if I remember correctly.

> Now that github actions is up and running maybe we should just ditch all the travis and ssh-key docs?

As long as we don't have #1177, we can't ditch the deploy key stuff. We could have GitHub Actions as the recommended setup (even as we're waiting for #1177), but no reason to ditch Travis -- it can still be useful for some people.

**Comment by bramtayl** [comment]
Close in favor of #1244

**Pull request closed without merging**